src/baobab.d.ts

```diff
@@ -0,0 +1,205 @@
+import Emitter from "emmett";
+
+interface PlainObject<T = any> {
+  [key: string]: T;
+}
+
+type Predicate = (data: any) => boolean;
+type Constraints = object;
+type PathKey = string | number;
+type PathElement = PathKey | Predicate | Constraints;
+export type Path = PathElement[] | PathKey;
+
+type Splicer = [number | object | ((...args: any[]) => any), ...any[]];
+
+/**
+ * This class is empty purposely. Baobab must be able to identify in an initial
+ * state when it has to deal with Monkeys instanciation, and uses this dummy
+ * class in that purpose.
+ */
+export class MonkeyDefinition {
+  // Empty class intended
+}
+
+export class Monkey {
+  // TODO
+}
+
+export interface BaobabOptions {
```

Comment:
Why is this an interface and not an object type? I always struggle to understand the difference. If using an interface doesn't this mean that someone will be required to pass an object containing all the keys?

Reply:
I don't know when using an interface or an object type as well... But in this case, only the `options` property of `Baobab` instances will require having all the keys (which makes sense to me, after having seen the implementation), while the options given to the constructor must match `Partial<BaobabOptions>` ([here](https://github.com/jacomyal/baobab/blob/afb33e03a41601c75a2f4ab055aa27c78bfc1e0d/src/baobab.d.ts#L156)).